### PR TITLE
Declutter CI/CD logs.

### PR DESCRIPTION
**Pull request by bwoodsend**
My war against noisy build logs:

- Move flaky testing to CI only. And disable flaky reporting which doesn't tell us anything we don't already know.
- Demote noisy logging of excluded imports to DEBUG.
- Drop pytest's log level to INFO.

I imagine we'll want to squash these but I've kept them separate in case anyone is particularly opposed to one of them.

### CODE REVIEW

**Comment by bwoodsend** [comment]
Would anyone object if I ignored the linter on this one. This is what it wants me to do:

```python
        default_args = ['--debug=bootloader', '--noupx',
                '--specpath', self._specdir,
                '--distpath', self._distdir,
                '--workpath', self._builddir,
                '--path', _get_modules_dir(self._request),
+                        '--log-level=INFO',
                ]
```

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -60,4 +60,5 @@ jobs:
 
       - name: Run tests
         run: |
-          pytest -n 3 --maxfail 3 --durations 10 tests/unit tests/functional --ignore tests/functional/test_libraries.py
+          pytest -n 3 --maxfail 3 --durations 10 tests/unit tests/functional --ignore tests/functional/test_libraries.py \
+            --force-flaky --no-success-flaky-report
```
Comment (rokm) [inline .github/workflows/macos.yml:64]: Should this be `--no-flaky-report` instead of `--no-success-flaky-report`? (Or both, everywhere?)

Comment (bwoodsend) [reply .github/workflows/macos.yml:64]: So it should...

**Comment by bwoodsend** [comment]
Hmm, the [django test mysteriously died](https://github.com/pyinstaller/pyinstaller/runs/2221424912?check_suite_focus=true#step:5:1642) the run before but worked this time. Looks like something gruesome is happening under there...

**Comment by bwoodsend** [comment]
@rokm @Legorooj Any last minute objections?

**Comment by rokm** [comment]
> @rokm @Legorooj Any last minute objections?

Nope.

**Review by Legorooj — APPROVED** [review]

**Pull request merged**